lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Backup.psm1

```diff
@@ -0,0 +1,24 @@
+# Copyright (c): 2018, Dag Wieers (@dagwieers) <dag@wieers.com>
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+Set-StrictMode -Version 2.0
+$ErrorActionPreference = "Stop"
+
+Function Backup-File($path, $obj=@{}, $check_mode=$false) {
+<#
+    .SYNOPSIS
+    Helper function to make a backup of a file.
+    .EXAMPLE
+    Backup-File -path $path -obj $result
+#>
+    $backup_path = "$path.$pid." + [DateTime]::Now.ToString("yyyyMMdd-HHmmss") + ".bak";
+    Try {
+        Copy-Item -Path $path -Destination $backup_path -WhatIf:$check_mode
+    } Catch {
+        Fail-Json -obj $obj -message "Failed to create backup file '$backup_path' from '$path'. ($($_.Exception.Message))"
+    }
+    return $backup_path
+}
+
+# This line must stay at the bottom to ensure all defined module parts are exported
+Export-ModuleMember -Alias * -Function * -Cmdlet *
```

Comment:
Prefer to be explicit on the exports instead of using a wildcard, this can be changed to `Export-ModuleMember -Function Backup-File`

Reply:
Ok, I think this was mostly copy-and-paste from elsewhere so most likely it's been wrong elsewhere too.

Reply:
Yea there are still some that use the wildcards which we should change.